azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDao.java

```diff
@@ -103,6 +108,36 @@ public void uploadExecutableFlow(final ExecutableFlow flow)
     }
   }
 
+  /**
+   * Select flows that are not in finished status, including both dispatched and non-dispatched
+   * flows.
+   *
+   * @return unfinished flow exec_ids
+   * @throws ExecutorManagerException the executor manager exception
+   */
+  List<Integer> selectUnfinishedFlows() throws ExecutorManagerException {
+    try {
+      return this.dbOperator.query(SelectFromExecutionFlows.SELECT_EXECUTION_BASE_QUERY
+              + "WHERE status NOT IN ("
```

Comment:
Let's also make sure that those selections have good index backup. execution_flows is a huge table, even if we're applying where clause but without index, it's still killing slow. 
I assume most of them had indexes already, but just to be safe to run a double check before merge, especially those time based columns.

Reply:
DONE. Added a submit_time index